src/v/datalake/parquet_writer.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "bytes/iobuf.h"
+
+#include <arrow/type.h>
+#include <parquet/arrow/writer.h>
+
+namespace datalake {
+
+namespace {
```

Comment:
mm, im not sure it makes sense to have an anonymous namespace in a header.

Reply:
+1 I think the code should work without wrapping the foward decl?